### DISCUSSION TITLE
Why doesn't this work? (compute series, assign += and -=)

### DISCUSSION THREAD

**Discussion by uvtc** [janet-lang/janet#904] [General]
Sorry for the vague question. I was trying to compute a series like this (let's see if LaTeX works in this forum):

$$\frac{1}{1} - \frac{1}{3} + \frac{1}{5} - \frac{1}{7} + \frac{1}{9} \cdots = \frac{\pi}{4}$$

Hm. Well, it's the series: 1/1 - 1/3 + 1/5 - 1/7 + 1/9 ... = pi/4

Firstly, I see that I can do it with:

~~~janet
(var res 0)
(var flag false)
(loop [i :range [1 100] :when (odd? i)]
  (if flag
    (set flag false)
    (set flag true))
  (if flag
    (+= res (/ 1 i))
    (-= res (/ 1 i))))

(print res)
(print (/ math/pi 4))
~~~

but it seemed like I could eliminate an `if`, so I tried this:

~~~janet
(var res 0)
(var pm-eq -=)
(loop [i :range [1 100] :when (odd? i)]
  (if (= pm-eq -=)
    (set pm-eq +=)
    (set pm-eq -=))
  (pm-eq res (/ 1 i)))

(print res)
(print (/ math/pi 4))
~~~

but `res` just stays zero... Why does this one not update `res`?

**Comment by sogaiu** [comment]
May be because `+=` and `-=` are macros and not functions?

On the note of reducing things, may be this sort of thing could work too?
```
(var res 0)
(loop [i :range [1 100] :when (odd? i)]
  (case (mod i 4)
    1
    (+= res (/ 1 i))
    3
    (-= res (/ 1 i))))

(print res)
(print (/ math/pi 4))
```

**Reply by uvtc** [reply]
> May be because += and -= are macros and not functions?

Ah. Right. Thanks!

And thanks for the additional code. That's a great observation that `(mod i 4)` always returns 1 or 3 given increasing consecutive odd numbers!
